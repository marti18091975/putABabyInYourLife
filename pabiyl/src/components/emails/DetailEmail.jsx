import React, { useState, useEffect } from "react";
import emailStore from "../../stores/emailStore";
import { loadEmails } from "../../actions/emailActions";

import "./detailEmail.css";

function DetailEmail(props) {
  const [emails, setEmails] = useState(emailStore.getEmails());
  const [detailElement, setDetailElement] = useState({});

  useEffect(() => {
    emailStore.addChangeListener(onChange);

    let emailId = props.match.params.emailId;
    if (emails.length === 0) {
      loadEmails();
    } else if (emailId) {
      const email = emailStore.getEmailById(emailId);
      if (email) {
        setDetailElement(email);
        console.log("&&&&&&&", detailElement);
      }
    }
    return () => emailStore.removeChangeListener(onChange);
  }, [onChange]);

  function onChange() {
    setEmails(emailStore.getEmails());
  }

  return (
    <main className="main__detail--email">
      <div className="email__container--detail">
        <div className="name__email text__email">
          from: {detailElement.emisorName}
        </div>
        <div className="text__email">date: {detailElement.date}</div>
        <div className="text__email time--email">
          hour: {detailElement.hour} : {detailElement.minute}
        </div>
        <div className="missage__email ">{detailElement.bodyMissage}</div>
      </div>
    </main>
  );
}
export default DetailEmail;
