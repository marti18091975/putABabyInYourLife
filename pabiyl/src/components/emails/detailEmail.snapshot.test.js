import renderer from 'react-test-renderer';
import React from 'react';
import DetailEmail from './DetailEmail';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {
    const props = {
        match: {
            params: {
                _id: "5f5dcd1985b8254bf0e90f23"
            }
        }
    };
    const tree = renderer.create(
        <Router>
            <DetailEmail {...props} />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});