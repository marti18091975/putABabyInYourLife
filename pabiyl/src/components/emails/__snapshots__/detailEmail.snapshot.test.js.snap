// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<main
  className="main__detail--email"
>
  <div
    className="email__container--detail"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="name__email text__email"
      >
        from: 
      </div>
    </a>
    <div
      className="text__email"
    >
      date: 
    </div>
    <div
      className="text__email time--email"
    >
      hour: 
       : 
    </div>
    <div
      className="missage__email "
    />
  </div>
</main>
`;
